tests/conftest.py

```diff
@@ -270,6 +276,7 @@ def https_server(cert_pem_file, cert_private_key_file):
         lifespan="off",
         ssl_certfile=cert_pem_file,
         ssl_keyfile=cert_private_key_file,
+        host="localhost",
```

Comment:
This is a bit of a less-than-ideal workaround for cert validation 😕. Since the server URL that the client uses is built using this host value (maybe it shouldn't be?) it needs to match the certificate name.